### PR TITLE
refactor: upgrade to support Golang 1.22

### PR DESCRIPTION
**Pull request by iutx**
#### What this PR does / why we need it:
support golang 1.22
1. Upgrade etcd client version.
2. Upgrade protobuf, grpc version.
3. Upgrade otel sdk version.
4. Update deprecated ioutil calls.

Used by project [playground/go-apm-demo](https://erda.cloud/playground/dop/projects/680/apps/12887/repo/tree/develop/go.mod).

#### Which issue(s) this PR fixes:
Fixes #

#### Specified Reviewers:
/assign @sfwn @chengjoey

### CODE REVIEW

**Review by chengjoey — APPROVED** [review]
Please also update `tools/dockerfile` and create a new gohub image

**Comment by iutx** [comment]
> Please also update `tools/dockerfile` and create a new gohub image

done :)
Working in process, a unit test failed beacause of etcd version upgraded.

**Comment by iutx** [comment]
I think we can work next stage, we need a reasonable amount of time to complete all 1.22 upgrades, not only `erda-infra` but also `erda`.
PTAL :) @chengjoey @sfwn

**Comment by chengjoey** [comment]
> I think we can work next stage, we need a reasonable amount of time to complete all 1.22 upgrades, not only `erda-infra` but also `erda`. PTAL :) @chengjoey @sfwn

[erda-upgrade](https://github.com/erda-project/erda/pull/6363)

**Review by chengjoey — APPROVED** [review]

File: go.sum [hunk 1]
```diff
@@ -1,8 +1,1315 @@
 bou.ke/monkey v1.0.2 h1:kWcnsrCNUatbxncxR/ThdYqbytgOIArtYWqcQLQzKLI=
 bou.ke/monkey v1.0.2/go.mod h1:OqickVX3tNx6t33n1xvtTtu85YN5s6cKwVug+oHMaIA=
 cloud.google.com/go v0.34.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
-cloud.google.com/go v0.38.0 h1:ROfEUZz+Gh5pa62DJWXSaonyu3StP6EA6lPEXPI6mCo=
 cloud.google.com/go v0.38.0/go.mod h1:990N+gfupTy94rShfmMCWGDn0LpTmnzTp2qbd1dvSRU=
+cloud.google.com/go v0.44.1/go.mod h1:iSa0KzasP4Uvy3f1mN/7PiObzGgflwredwwASm/v6AU=
```
Comment (sfwn) [inline go.sum:5]: so much versions of a same pkg. do you run `go mod tidy`?

**Review by sfwn — CHANGES_REQUESTED** [review]

File: go.sum (hunk 1 above)
Comment (iutx) [reply go.sum:5]: sure :)  that is a necessary action.
All module in used.
```shell
➜  erda-infra git:(feat/support-go1.22) ✗ go mod tidy -v
➜  erda-infra git:(feat/support-go1.22) ✗ 
```
I try to remove `go.sum` and re-created, nothing changed within module `cloud.google.com/go`.

Comment (iutx) [reply go.sum:5]: It seems that now `go.sum` changed with re-create, PTAL.

File: go.mod [hunk 2]
```diff
@@ -64,92 +66,97 @@ require (
...
 
 replace (
-	github.com/coreos/bbolt v1.3.5 => go.etcd.io/bbolt v1.3.5
-	go.etcd.io/bbolt v1.3.5 => github.com/coreos/bbolt v1.3.5
-	go.opentelemetry.io/otel => go.opentelemetry.io/otel v1.17.0
-	go.opentelemetry.io/otel/metric => go.opentelemetry.io/otel/metric v1.17.0
-	go.opentelemetry.io/otel/sdk => go.opentelemetry.io/otel/sdk v1.2.0
-	go.opentelemetry.io/otel/trace => go.opentelemetry.io/otel/trace v1.2.0
-	google.golang.org/grpc => google.golang.org/grpc v1.26.0
+	google.golang.org/grpc => github.com/grpc/grpc-go v1.63.2
```
Comment (iutx) [inline go.mod:159]: Verified, cause by this change.
I will try to checkout it.

**Review by iutx — COMMENTED** [review]
@sfwn wip.

**Comment by iutx** [comment]
I forgot why I did the operation :(, please pay attention when erda upgrading. @chengjoey

- [ref](https://stackoverflow.com/questions/69825533/why-does-go-sum-include-so-many-older-packages)
- Tested on [go-apm-demo mr](https://erda.cloud/playground/dop/projects/680/apps/12887/repo/commit/113a800b637e001e9c3dd2ff2e8e153980f17f8a).

File: go.mod (hunk 2 above)
Comment (iutx) [reply go.mod]: done. 
PTAL @sfwn

**Comment by iutx** [comment]
I think all job done, please take a look. @sfwn

**Review by sfwn — APPROVED** [review]

**Pull request merged**